File: NetFramework/XHtmlEditor/VB/CustomDialogs/Controls/ReplaceOptions.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Friend Partial Class ReplaceOptions
	Inherits FindOptions
	Public Sub New()
		MyBase.New()
		InitializeComponent()

        AddHandler _btnReplace.Click, AddressOf _btnReplace_Click

        AddHandler _btnReplaceAll.Click, AddressOf _btnReplaceAll_Click

        _tbFindPattern_TextChanged(Me, EventArgs.Empty)
    End Sub

    Sub _btnReplace_Click(ByVal sender As Object, ByVal e As EventArgs)
        RaiseButtonReplaceClick()
    End Sub

    Sub _btnReplaceAll_Click(ByVal sender As Object, ByVal e As EventArgs)
        RaiseButtonReplaceAllClick()
    End Sub

    Private Sub _tbFindPattern_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
        _btnReplace.Enabled = _btnFindNext.Enabled
        _btnReplaceAll.Enabled = _btnFindNext.Enabled
    End Sub

#Region "events"

    Public Event ButtonReplaceClick As EventHandler

    Protected Sub RaiseButtonReplaceClick()
        RaiseEvent ButtonReplaceClick(Me, EventArgs.Empty)
    End Sub

    Public Event ButtonReplaceAllClick As EventHandler

    Protected Sub RaiseButtonReplaceAllClick()
        RaiseEvent ButtonReplaceAllClick(Me, EventArgs.Empty)
    End Sub

#End Region

#Region "properties"

    Public ReadOnly Property ReplacePattern() As String
        Get
            Return _tbReplacePattern.Text.Trim()
        End Get
    End Property

#End Region
End Class

File: NetFramework/XHtmlEditor/CS/CustomDialogs/Controls/ReplaceOptions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CustomDialogs
{
    internal partial class ReplaceOptions : FindOptions
    {
        public ReplaceOptions() : base()
        {
            InitializeComponent();

            _btnReplace.Click += delegate(object sender, EventArgs e) { RaiseButtonReplaceClick(); };
            _btnReplaceAll.Click += delegate(object sender, EventArgs e) { RaiseButtonReplaceAllClick(); };

            _tbFindPattern_TextChanged(this, EventArgs.Empty);
        }

        private void _tbFindPattern_TextChanged(object sender, EventArgs e)
        {
            _btnReplace.Enabled = _btnFindNext.Enabled;
            _btnReplaceAll.Enabled = _btnFindNext.Enabled;
        }

        #region events

        public event EventHandler ButtonReplaceClick;

        protected void RaiseButtonReplaceClick()
        {
            if (ButtonReplaceClick != null)
                ButtonReplaceClick(this, EventArgs.Empty);
        }

        public event EventHandler ButtonReplaceAllClick;

        protected void RaiseButtonReplaceAllClick()
        {
            if (ButtonReplaceAllClick != null)
                ButtonReplaceAllClick(this, EventArgs.Empty);
        }

        #endregion

        #region properties

        public string ReplacePattern
        {
            get { return _tbReplacePattern.Text.Trim(); }
        }

        #endregion
    }
}

